benchmarks/union_find/union_find.cpp

```diff
@@ -0,0 +1,259 @@
+        KOKKOS_LAMBDA(int i) {
+          auto rand_gen = rand_pool.get_state();
+          random_values(i) = rand_gen.rand();
+          rand_pool.free_state(rand_gen);
+        });
+    auto permute = ArborX::Details::sortObjects(exec_space, random_values);
+
+    // Init edges in a random order
+    Kokkos::parallel_for(
+        "ArborX::Bechmark::init_edges",
+        Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+        KOKKOS_LAMBDA(unsigned i) {
+          auto rand_gen = rand_pool.get_state();
+          edges(permute(i)) = {rand_gen.urand() % (i + 1), i + 1};
+          rand_pool.free_state(rand_gen);
+        });
+  }
+  return edges;
+}
+
+template <typename ExecutionSpace>
+auto buildUnionFind(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::View<int *, MemorySpace> labels(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::labels"),
+      n);
+  ArborX::iota(exec_space, labels);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if constexpr (std::is_same_v<ExecutionSpace, Kokkos::Serial>)
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ true>(labels);
+  else
+#endif
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ false>(labels);
+}
+
+template <typename ExecutionSpace>
+void BM_union_find(benchmark::State &state, Spec const &spec)
+{
+  ExecutionSpace exec_space;
+
+  auto const n = spec.num_edges + 1;
+
+  auto edges = buildEdges(exec_space, spec.num_edges, spec.allow_loops);
+  auto union_find = buildUnionFind(exec_space, n);
+
+  for (auto _ : state)
+  {
+    exec_space.fence();
+    auto const start = std::chrono::high_resolution_clock::now();
+
+    Kokkos::parallel_for(
+        "ArborX::Bechmark::union-find",
+        Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, edges.size()),
+        KOKKOS_LAMBDA(int e) {
+          int i = edges(e).source;
+          int j = edges(e).target;
+
+          union_find.merge(i, j);
+        });
+
+    exec_space.fence();
+    auto const end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    state.SetIterationTime(elapsed_seconds.count());
+  }
+  state.counters["rate"] = benchmark::Counter(
+      spec.num_edges, benchmark::Counter::kIsIterationInvariantRate);
+}
+
+// NOTE Motivation for this class that stores the argument count and values is
+// I could not figure out how to make the parser consume arguments with
+// Boost.Program_options
+// Benchmark removes its own arguments from the command line arguments. This
+// means, that by virtue of returning references to internal data members in
+// argc() and argv() function, it will necessarily modify the members. It will
+// decrease _argc, and "reduce" _argv data. Hence, we must keep a copy of _argv
+// that is not modified from the outside to release memory in the destructor
+// correctly.
+class CmdLineArgs
```

Comment:
This is common with `benchmarks/bvh_driver`. Should be moved into a separate file, but not sure where.